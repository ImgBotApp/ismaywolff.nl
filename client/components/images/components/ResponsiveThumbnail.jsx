import React, { PropTypes } from 'react'
import sizeMe from 'react-sizeme'
import Thumbnail from './Thumbnail'
import { roundUp } from '../utils'

export function ResponsiveThumbnail({ id, size }) {
  return (
    <Thumbnail
      id={id}
      width={roundUp(size.width)}
    />
  )
}

ResponsiveThumbnail.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.object.isRequired
}

export default sizeMe()(ResponsiveThumbnail)
