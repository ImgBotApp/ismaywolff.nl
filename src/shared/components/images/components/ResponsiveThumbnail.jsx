import React from 'react'
import { string } from 'prop-types'
import { AsyncReactMeasure } from '../../async'
import { getImageWidth } from '../utils'
import Thumbnail from './Thumbnail'

function ResponsiveThumbnail({ id }) {
  return (
    <AsyncReactMeasure includeMargin={false} whitelist={['width']}>
      {dimensions => <Thumbnail id={id} width={getImageWidth(dimensions.width)} />}
    </AsyncReactMeasure>
  )
}

ResponsiveThumbnail.propTypes = {
  id: string.isRequired
}

export default ResponsiveThumbnail
