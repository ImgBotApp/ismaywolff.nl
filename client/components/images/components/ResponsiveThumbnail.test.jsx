import React from 'react'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import ResponsiveThumbnail from './ResponsiveThumbnail'

jest.mock('./Thumbnail', () => (
  () => <div />
))

describe('<ResponsiveThumbnail />', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <ResponsiveThumbnail id={'one'} />
    )
    expect(mountToJson(wrapper)).toMatchSnapshot()
  })
})
