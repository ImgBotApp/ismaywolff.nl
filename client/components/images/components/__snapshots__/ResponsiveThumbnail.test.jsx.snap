// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveThumbnail /> renders correctly 1`] = `
<ResponsiveThumbnail
  id="one"
  size={
    Object {
      "width": 50,
    }
  }
>
  <Component
    id="one"
    width={50}
  >
    <div />
  </Component>
</ResponsiveThumbnail>
`;
